src/main/java/PlayerInteractor.java

```diff
@@ -0,0 +1,121 @@
+/** relevant imports */
+import java.util.ArrayList;
+
+/**
+ * PlayerInteractor class
+ * Responsible for updating the PlayerData class.
+ */
+public class PlayerInteractor {
+
+    /**
+     * The PlayerData object that this class will interact with.
+     */
+    PlayerData player;
+
+    /**
+     * Creates a PlaterInteractor object.
+     * @param player The Player that this object will interact with.
+     */
+    public PlayerInteractor(PlayerData player) {
+        this.player = player;
+    }
+
+    /**
+     * Updates the PlayerData object that this class interacts with.
+     * @param p the new PlayerData object.
+     */
+    public void updatePlayer(PlayerData p) {
+        player = p;
+    }
+
+    /**
+     * Updates the Player's health.
+     * @param val The amount of health to add.
+     */
+    public void updateHealth(int val) {
+        player.setHealth(player.getHealth() + val);
+    }
+
+    /**
+     * Returns the PlayerData object's health status.
+     * @return The health of the PlayerData object.
+     */
+    public int getPlayerHealth() {
+        return player.getHealth();
+    }
+
+    /**
+     * Updates the player's attack power.
+     * @param attackPower the value to be added to the PlayerData object.
+     */
+    public void updateAttackPower(int attackPower) {
+        player.setAttackPower(player.getAttackPower() + attackPower);
+    }
+
+    /**
+     * Returns the player's attack power.
+     * @return the PlayerData object's attack power.
+     */
+    public int getAttackPower() {
+        return player.getAttackPower();
+    }
+    /**
+     * Updates the event the player is on in the game.
+     * @param eventID The UUID of the event.
+     */
+    public void updateEvent(int eventID) {
+        player.setEventID(eventID);
+    }
+
+    /**
+     * Returns what event id is stored in the PlayerData object.
+     * @return The current event stored in the PlayerData object.
+     */
+    public int getPlayerEventID() {
+        return player.getEventID();
+    }
+
+    /**
+     * Sets the username of the PlayerData object.
+     * @param un The new username value.
+     */
+    public void setPlayerUsername(String un) {
```

Comment:
if we set the username to be final and don't allow the username to be changed, we can remove the setPlayerUsername method?